debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java

```diff
@@ -120,11 +120,30 @@ public Config validate(Map<String, String> connectorConfigs) {
                     }
                     // check user for LOGIN and REPLICATION roles
                     if (!connection.queryAndMap(
-                            "SELECT rolcanlogin, rolreplication FROM pg_roles WHERE rolname = current_user",
-                            connection.singleResultMapper(rs -> rs.getBoolean("rolcanlogin") && rs.getBoolean("rolreplication"), "Could not fetch roles"))) {
+                            "SELECT r.rolcanlogin AS rolcanlogin, r.rolreplication AS rolreplication," +
```

Comment:
How about using query like `SELECT rolcanlogin, rolreplication FROM pg_roles WHERE rolname = current_user UNION ALL SELECT rolcanlogin, rolreplication FROM pg_catalog.pg_auth_members m JOIN pg_catalog.pg_roles b ON (m.roleid = b.oid) WHERE oid=(SELECT oid FROM pg_roles WHERE rolname = current_user)` and check there is at least one `true` for each of the roles?

Reply:
Wouldn't this only work when we are on AWS, but then will fail when we are not? Had a hard time to figure a smaller & nicer, less repetitive query. Also, a proper solution needs to check a role tree (like a role is assigned to another role and another role and this role finally to the user). This should be handled algorithmically and not with a complex query with https://issues.redhat.com/browse/DBZ-2894 .

Reply:
It should work for both, that's while there is the `UNION`. IMHO it just really does not cover the inheritance